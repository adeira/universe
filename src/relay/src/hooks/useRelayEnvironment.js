// @flow

import * as React from 'react';
import { ReactRelayContext } from 'react-relay';
import { invariant } from '@adeira/js';

import type { Environment } from '../runtimeTypes.flow';

export default function useRelayEnvironment(): Environment {
  const context = React.useContext(ReactRelayContext);
  invariant(
    context != null,
    'useRelayEnvironment: Expected to have found a Relay environment provided by' +
      'a `RelayEnvironmentProvider` component or `QueryRenderer`. This usually means that' +
      'useRelayEnvironment was used in a component that is not a descendant of these components. ' +
      'Please make sure a `RelayEnvironmentProvider` has been rendered somewhere as a parent of ' +
      'your component. It should be defined only once in the whole application.',
  );
  // $FlowFixMe errors after upgrading to relay 9.1.0
  return context.environment;
}
