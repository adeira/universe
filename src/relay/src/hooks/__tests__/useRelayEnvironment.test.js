// @flow

import { renderHook } from '@testing-library/react-hooks';

import useRelayEnvironment from '../useRelayEnvironment';

it('throws the correct message', () => {
  const { result } = renderHook(useRelayEnvironment);
  expect(() => result.current).toThrowErrorMatchingSnapshot();
});
