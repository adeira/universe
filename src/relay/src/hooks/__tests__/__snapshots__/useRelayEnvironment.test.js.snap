// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`throws the correct message 1`] = `"useRelayEnvironment: Expected to have found a Relay environment provided by a \`RelayEnvironmentProvider\` component or \`QueryRenderer\`. This usually means that \`useRelayEnvironment\` was used in a component that is not a descendant of these components. Please make sure a \`RelayEnvironmentProvider\` has been rendered somewhere as a parent of your component. It should be defined only once in the whole application."`;
